core/resume_hook.cc

```diff
@@ -72,13 +72,17 @@ void run_global_resume_hooks(bool run_modules) {
   }
 
   if (run_modules) {
+    std::set<Module *> erase_modules; // Erase modules after completing iteration
     auto &resume_modules = event_modules[Event::PreResume];
     for (Module *m : resume_modules) {
       int ret = m->OnEvent(Event::PreResume);
       if (ret == -ENOTSUP) {
-        resume_modules.erase(m);
+        erase_modules.insert(m);
```

Comment:
In both of these cases, might it be simpler to use iterators rather than colon notation and a second `set`?  e.g., `resume_modules.erase(it++);` should work fine here.

Reply:
(Oh, you, er, @barath, meant _ditch_ the range-based `for` loop.  I seem to be extra slow this morning...)

Reply:
OK, will use iterators. Thanks.